lib/TWCManager/TWCSlave.py

```diff
@@ -461,11 +462,24 @@ def receive_slave_heartbeat(self, heartbeatData):
         self.reportedAmpsActual = ((heartbeatData[3] << 8) + heartbeatData[4]) / 100
         self.reportedState = heartbeatData[0]
 
+        if (self.reportedAmpsActual != self.reportedAmpsLast):
```

Comment:
By moving it here, you're bypassing the "significant change" logic. Is that intentional? If so, why?

Reply:
I agree. To be honest - I saw this a week ago, but it did not work multiple times. I had always a value there for more the whole night, even without charging. By the way in the HASSStatus class there is already a limitation for too much updating. I will have another look here...

Reply:
In my opinion it should exactly be updated there and not afterwards. The overload of status reporting should be handled in the Status classes. To be clear - I had it several times during night - that charging was over and it still shows me power... Now it works like a charm